extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -116,6 +117,19 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Resolves relative URLs to their full path.
+   * @param {string} url
+   * @private
+   * @return {string}
+   */
+  resolveRelativeUrl_(url) {
+    if (/^\//.test(url) && isSecureUrlDeprecated(url)) {
+      return resolveRelativeUrl(url, this.getAmpDoc().getUrl());
```

Comment:
See comment on `getUrl()`:

https://github.com/ampproject/amphtml/blob/020727b15d1e78ffe8fd8021009bdd057de6a00a/src/service/ampdoc-impl.js#L524-L528

Using the URL from which the document was loaded actually causes the issue with cache URLs. You want the `sourceUrl`:

```suggestion
      const {sourceUrl} = Services.documentInfoForDoc(element);
      return resolveRelativeUrl(url, sourceUrl);
```